### ISSUE TITLE
Prefix setting on un-matched fields affects other field matches

### ISSUE THREAD

**Issue by Sembiance** [typesense/typesense#632]
When running typesense `0.23.0` on `linux 5.17.1` changing the `prefix` value on an un-matched field causes odd behavior.

Exact reproducible steps below, including test data.

Create our data dir and run our server:
```
mkdir data
typesense-server --data-dir=./data --api-address=127.0.0.1 --api-port=8108 --peering-address=127.0.0.1 --api-key=TESTCASEKEY
```

Create the schema:
```
curl "http://localhost:8108/collections" -X POST -H "Content-Type: application/json" -H "X-TYPESENSE-API-KEY: TESTCASEKEY" -d '{
"name" : "twig",
"fields" : [
{ "name" : "filename", "type" : "string" },
{ "name" : "size", "type" : "int64" },
{ "name" : "nsfw", "type" : "int32" },
{ "name" : "itemid", "type" : "int32", "facet" : true },
{ "name" : "catid", "type" : "int32", "facet" : true },
{ "name" : "genreid", "type" : "int32", "facet" : true },
{ "name" : "family", "type" : "string", "facet" : true },
{ "name" : "formatid", "type" : "string", "facet" : true, "optional" : true },
{ "name" : "ts", "type" : "int64", "optional" : true },
{ "name" : "content", "type" : "string", "optional" : true },
{ "name" : "ext", "type" : "string", "optional" : true },
{ "name" : "label", "type" : "string[]", "optional" : true },
{ "name" : "duration", "type" : "int32", "optional" : true },
{ "name" : "width", "type" : "int32", "optional" : true },
{ "name" : "height", "type" : "int32", "optional" : true },
{ "name" : "animated", "type" :  "bool", "optional" : true },
{ "name" : "title", "type" : "string", "optional" : true },
{ "name" : "subCount", "type" : "int32", "optional" : true, "index" : false },
{ "name" : "href", "type" : "string", "optional" : true, "index" : false },
{ "name" : "contentURL", "type" : "string", "optional" : true, "index" : false },
{ "name" : "extra", "type" : "string", "optional" : true, "index" : false }
] }'
```

Download the attached [2369.jsonl.zip](https://github.com/typesense/typesense/files/8976388/2369.jsonl.zip) and index it, containing 12,091 documents:
```
unzip 2369.jsonl.zip
curl --silent "http://127.0.0.1:8108/collections/twig/documents/import" -X POST -H "X-TYPESENSE-API-KEY: TESTCASEKEY" --data-binary @"2369.jsonl" | sort | uniq -c
```

Perform a query with `prefix=false`:
```
curl --silent -H "X-TYPESENSE-API-KEY: TESTCASEKEY" 'http://127.0.0.1:8108/collections/twig/documents/search?split_join_tokens=false&prioritize_exact_match=true&num_typos=0&typo_tokens_threshold=0&drop_tokens_threshold=0&q=jungle+girl&query_by=content&prefix=false'
```

It finds 1 document, matching the queried value in the `content` field.

Now modify the `query_by` adding the field `filename` and keeping `prefix=false`:
```
curl --silent -H "X-TYPESENSE-API-KEY: TESTCASEKEY" 'http://127.0.0.1:8108/collections/twig/documents/search?split_join_tokens=false&prioritize_exact_match=true&num_typos=0&typo_tokens_threshold=0&drop_tokens_threshold=0&q=jungle+girl&query_by=content,filename&prefix=false'
```

As expected, it still finds that 1 document, matched in the `content` field.

Now modify the `prefix` field so `content` remains `false` but `filename` is now `true`:
```
curl --silent -H "X-TYPESENSE-API-KEY: TESTCASEKEY" 'http://127.0.0.1:8108/collections/twig/documents/search?split_join_tokens=false&prioritize_exact_match=true&num_typos=0&typo_tokens_threshold=0&drop_tokens_threshold=0&q=jungle+girl&query_by=content,filename&prefix=false,true'
```

Now the document is no longer matched. Yes, if I set `max_candidates=1000` then the document is matched.

However since nothing is matched to the `filename` field, I wouldn't expect modifying the `prefix` setting of that unrelated field to affect finding the document.

Is this a bug, limitation or intended behavior?

**Comment by kishorenc** [comment]
I'll be looking into this soon. Will update.

**Comment by Sembiance** [comment]
> I'll be looking into this soon. Will update.

No rush. I've decided to just turn off prefix matching on all fields, which works around the issue nicely.

**Comment by kishorenc** [comment]
Fixed in `0.24.0.rc12`.
